ecs/cloudformation.go

```diff
@@ -675,32 +675,53 @@ func (b *ecsAPIService) createPolicies(project *types.Project, service types.Ser
 						},
 					},
 				},
-				PolicyName: fmt.Sprintf("%sGrantAccessToSecrets", service.Name),
+				PolicyName: normalizeResourceName("%sGrantAccessToSecrets", service.Name),
 			},
 		}
 	}
 	return nil
 }
 
 func networkResourceName(network string) string {
-	return fmt.Sprintf("%sNetwork", normalizeResourceName(network))
+	return normalizeResourceName("%sNetwork", network)
 }
 
 func serviceResourceName(service string) string {
-	return fmt.Sprintf("%sService", normalizeResourceName(service))
+	return normalizeResourceName("%sService", service)
 }
 
 func volumeResourceName(service string) string {
-	return fmt.Sprintf("%sFilesystem", normalizeResourceName(service))
+	return normalizeResourceName("%sFilesystem", service)
 }
 
-func normalizeResourceName(s string) string {
-	chk := fmt.Sprintf("%08x", crc32.ChecksumIEEE([]byte(s)))
-	ts := strings.Title(regexp.MustCompile("[^a-zA-Z0-9]+").ReplaceAllString(s, ""))
-	if len(ts) > 6 {
-		ts = ts[:6]
+// normalizeResourceName will remove invalid template element name characters,
+// create a CamelCase style in resulting string and add 4 bytes of the string CRC
```

Comment:
I understand the intent here but I don't like we polute the logical resource name and make them unpredictable (for a humain brain, which can't compute CRCs :P)
I don't expect conflicts to be such a common thing, and we could detect those before conversion

Reply:
I had a lot of naming collisions when adding path based rules because the path names are too long for being used as the resource name without truncating it, and by truncating/removing non alphabetic chars the chance of a collision was high. We could enhance this mechanism by verifying if there is, among all rules, a "real" collision and then add a counter to the resource name when adding path based routing to ECS here...